src/Features/Core/Portable/ExternalAccess/UnitTesting/API/NewUnitTestingIncrementalAnalyzer.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api;
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.SolutionCrawler;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api;
+
+internal sealed partial class NewUnitTestingIncrementalAnalyzerProvider
+{
+    private class NewUnitTestingIncrementalAnalyzer : IUnitTestingIncrementalAnalyzer
+    {
+        private readonly INewUnitTestingIncrementalAnalyzerImplementation _implementation;
+
+        public NewUnitTestingIncrementalAnalyzer(INewUnitTestingIncrementalAnalyzerImplementation implementation)
+            => _implementation = implementation;
+
+        public Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeDocumentAsync(document, bodyOpt, reasons, cancellationToken);
+
+        public Task AnalyzeProjectAsync(Project project, bool semanticsChanged, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeProjectAsync(project, semanticsChanged, reasons, cancellationToken);
+
+        public Task AnalyzeSyntaxAsync(Document document, UnitTestingInvocationReasons reasons, CancellationToken cancellationToken)
+            => _implementation.AnalyzeSyntaxAsync(document, reasons, cancellationToken);
+
+        public Task DocumentCloseAsync(Document document, CancellationToken cancellationToken)
+            => Task.CompletedTask;
```

Comment:
it's clear unit testing doesn't need a lot of the functionality of the full solution crawler.  i would like to start if-def'ing out these entire areas that aren't used to help make the amount of code/functionality that is supported here even smaller.